### ISSUE TITLE
Fast partitioned sum could be more general

### ISSUE THREAD

**Issue by rikedyp** [abrudz/aplcart#101]
The fast partitioned sum does not return the same results as `+/¨⍺⊂⍵` if `⍺` does not start with a `1`

```APL
      0 1 0 1 0 0 0 {¯2-⌿0⍪(1⌽⍺)⌿+⍀⍵} 1 3 2 4 4 4 4
1 5
      0 1 0 1 0 0 0 {+/¨⍺⊂⍵} 1 3 2 4 4 4 4
5 16
```

This can be rectified with some conditional modification of the partition vector

```APL
      0 1 0 1 0 0 0 {¯2-⌿(0/⍨⊃⍺)⍪(1@(≢⍺)⊢1⌽⍺)⌿+⍀⍵} 1 3 2 4 4 4 4
5 16
```

**Comment by abrudz** [comment]
`{¯2-⌿(0/⍨⊃⍺)⍪(1⌽1@1⊢⍺)⌿+⍀⍵}` is better but it is very rare to actually need this, and it makes the code much more involved (though at next to no performance cost). I think it better to stipulate that all the fast partitioned algorithms require `Av` beginning with `1`.

**Comment by rikedyp** [comment]
Yeah I agree, the original is much simpler. Might be nice to add comments to aplcart examples.  
Here is a prototype format: APLCart description in the header, main example in code, comments in the footer:

[TiO flat partitioned sum example](https://tio.run/##hZKxbtswEIb3PMVtteEGttJ0b14gS57gLJ1sIjTliJQcIShQtEAHIyq6dOzSKeiSKUuAIkAehS/iHk8KI3sJAUE66r/jx/sP1/o4a1AXi51vf8NFtYLRZDqGIgfCdAm2mteUuqIMO@egTKZSdJTBvIGzGkY5WgeT6fPdWe23X8/r8c7/@AX@@09IYCbPDG6e70@O/e3TzLd/R4m//efbxzHHE99@8e3DZ5adwAc4hY/CkFg@BlgDKywvwS0JrMPSBYI1v5VThbFHQVqSrbQDZTsVww81TMnlRfgWgoj2Vv@Hv7JCqqfVqtLoVC0HDROGBbsdunYlpi7kWZFbQImATBZ7G0FfiwW@IcUl0VryatQVhcwQ5KrkrkeK/m77/GxcZAiBpauKjDu4BrgCFhRE3MPI885KcamupL0IYnRNpeX/geMmmM6XZtfFw8LoBjZL6pwTwyxslFtybvJSKUUDc@ITDZWole3maFVkKm@UWQhrpIBu8I7CGPWDdHBmnJuhpu/GaDb17Z3ffhNjeteTT4nf/nnT@7j2cjgOStQbbCxUbGznhUF9OBH9GiDs7Us38mGjkvevVkd3d/8B)
